cmd/rm-main.go

```diff
@@ -121,17 +133,34 @@ func checkRmSyntax(ctx *cli.Context) {
 	isForce := ctx.Bool("force")
 	isRecursive := ctx.Bool("recursive")
 	isStdin := ctx.Bool("stdin")
-
+	isDangerous := ctx.Bool("dangerous")
+	isNamespaceRemoval := false
+	for _, url := range ctx.Args() {
+		// namespace removal applies only for non FS. So filter out if passed url represents a directory
+		if !isAliasURLDir(url) {
+			_, path := url2Alias(url)
```

Comment:
Does `url2Alias()` take care of returning clean path including windows UNC path?

Reply:
yes, see test output below..
```
C:\go\bin> cd ..\src\github.com\minio\mc
C:\go\src\github.com\minio\mc [rr2 ≡]> mc rm --r --force \\myserver\myfolder\myfile
mc.exe: <ERROR> Failed to remove `\\myserver\myfolder\myfile` recursively. open \\myserver\myfolder\myfile: The network name cannot be found.
C:\go\src\github.com\minio\mc [rr2 ≡]> mc rm --r --force C:
Removing `C:.git\FETCH_HEAD`.

C:\go\src\github.com\minio\mc> mc rm C:\
Removing `C:\`.
mc.exe: <ERROR> Failed to remove `C:\`. Insufficient permissions to access this file `C:\`
C:\go\src\github.com\minio\mc>
C:\go\src\github.com\minio\mc> mc rm C:
Removing `C:`.
mc.exe: <ERROR> Failed to remove `C:`. remove C:: The process cannot access the file because it is being used by a
ther process.
C:\go\src\github.com\minio\mc>

C:\go\src\github.com\minio\mc> mc rm --r --force  C:
Removing `C:cmd\mc_test.go`.

~> mc mb a/b/c/d
~> mc rm a
Removing `a`.
mc.exe: <ERROR> Failed to remove `a`. remove a: The directory is not empty.
~> mc rm \\TMP\ss\
mc.exe: <ERROR> Failed to remove `\\TMP\ss\`. GetFileAttributesEx \\TMP\ss: The network name cannot be
~> mc rm C:\Users\vagrant\a\\\
Removing `C:\Users\vagrant\a\\\`.
mc.exe: <ERROR> Failed to remove `C:\Users\vagrant\a\\\`. remove C:\Users\vagrant\a\\\: The directory
~> mc rm C:\Users\vagrant\a\\\b\\\c\\\d
Removing `C:\Users\vagrant\a\\\b\\\c\\\d`.
~>
```

Reply:
We used `path.Clean()` to take care aliases like `s3` and `s3/`.  Hope return `path` from `url2Alias()` is clean.

Reply:
yes, added a comment for that as well.